src/Job/DeveloperSync.php

```diff
@@ -59,128 +58,128 @@ class DeveloperSync extends EdgeJob {
+    $users = [];
+    /** @var \Drupal\user\UserInterface $user */
+    foreach (User::loadMultiple() as $user) {
+      $email = $user->getEmail();
+      if (isset($email)) {
+        $users[strtolower($email)] = $user;
       }
     }
 
-    return $accounts;
+    return $users;
   }
 
   /**
-   * Loads all Apigee Edge developers' emails.
+   * Loads all Apigee Edge developers indexed my their emails.
    *
-   * @return array
-   *   Format: strtolower(email) => email
+   * @return \Drupal\apigee_edge\Entity\DeveloperInterface[]
+   *   Format: strtolower(email) => Developer
    */
-  protected function loadEdgeUserEmails() : array {
-    $controller = new DeveloperController($this->getConnector()->getOrganization(), $this->getConnector()->getClient());
-    $mails = $controller->getEntityIds();
-
-    $accounts = [];
-    foreach ($mails as $mail) {
-      if ($this->filter && !preg_match($this->filter, $mail)) {
+  protected function loadDevelopers(): array {
+    $developers = [];
+    /** @var \Drupal\apigee_edge\Entity\DeveloperInterface $developer */
+    foreach (Developer::loadMultiple() as $developer) {
+      $email = $developer->getEmail();
+      if ($this->filter && !preg_match($this->filter, $email)) {
         continue;
       }
-      $accounts[strtolower($mail)] = $mail;
+      else {
+        $developers[strtolower($email)] = $developer;
+      }
     }
 
-    return $accounts;
+    return $developers;
   }
 
   /**
    * {@inheritdoc}
    */
   protected function executeRequest() {
-    $this->drupalAccounts = $this->loadUserEmails();
-    $this->edgeAccounts = $this->loadEdgeUserEmails();
+    $this->drupalUsers = $this->loadUsers();
+    $this->edgeDevelopers = $this->loadDevelopers();
   }
 
   /**
    * {@inheritdoc}
    */
-  public function execute() : bool {
+  public function execute(): bool {
     parent::execute();
 
-    foreach ($this->edgeAccounts as $search => $mail) {
-      if (empty($this->drupalAccounts[$search])) {
-        $createUserJob = new CreateUser($mail);
+    // Update Apigee Edge developers and Drupal users if needed.
+    $identical_entities = array_intersect_key($this->edgeDevelopers, $this->drupalUsers);
+    foreach ($identical_entities as $clean_email => $entity) {
+      /** @var \Drupal\apigee_edge\Entity\DeveloperInterface $developer */
+      $developer = $this->edgeDevelopers[$clean_email];
+      /** @var \Drupal\user\UserInterface $user */
+      $user = $this->drupalUsers[$clean_email];
+
+      $last_modified_delta = $developer->getLastModifiedAt()->getTimestamp() - $user->getChangedTime();
+      // Update Drupal user because the Apigee Edge developer is fresher.
+      if ($last_modified_delta >= 0) {
+        $updateUserJob = new UserUpdate($user->getEmail());
+        $updateUserJob->setTag($this->getTag());
+        $this->scheduleJob($updateUserJob);
+      }
+      // Update Apigee Edge developer because the Drupal user is fresher.
```

Comment:
https://www.urbandictionary.com/define.php?term=fresher ? :) Is "more recent" for example.